java/dataset/src/main/java/org/apache/arrow/dataset/jni/NativeScanner.java

```diff
@@ -68,6 +68,15 @@ ArrowReader execute() {
   }
 
   @Override
+  public NativeScanTask scanTask() {
```

Comment:
We just shouldn't have ScanTask at all, there is no such thing on the C++ side anymore.

Reply:
Ok, let me delete that, this `scan()` method is exposed to the client and will cause break compatibilities

Reply:
The deprecation makes sense to me, I think we should consider what the new interface is, though. I would think something like `ArrowReader scanBatchesUnordered()`.

Reply:
That would mirror the C++ API

Reply:
Changed